lib/benchee/benchmark/runner.ex

```diff
@@ -146,12 +138,15 @@ defmodule Benchee.Benchmark.Runner do
                       current_time: current_time, end_time: end_time
                     }) when current_time > end_time do
     # restore correct order - important for graphing
-    %Scenario{scenario | run_times: Enum.reverse(run_times)}
+    %Scenario{scenario | run_times: run_times |> List.flatten |> Enum.sort}
   end
   defp do_benchmark(scenario = %Scenario{run_times: run_times},
                     scenario_context) do
-    run_time = iteration_time(scenario, scenario_context)
-    updated_scenario = %Scenario{scenario | run_times: [run_time | run_times]}
+    new_run_times = Parallel.map(0..scenario_context.config.parallel, fn _ ->
```

Comment:
I moved the parallelization down to the actual collection of the runtimes. I think I actually rather like this - good call! Let me know what you think, though.

Reply:
I think this is might be too far down :) Here for each iteration we'd spin up new processes, which is cheap yes but we wanna be super fast so it might not be cheap enough :D 

I was thinking about a place like [here](https://github.com/PragTob/benchee/blob/master/lib/benchee/benchmark/runner.ex#L40-L42) more or less. That'd need some changes though. I.e. we wouldn't carry along the scenario in `measure_runtimes` anymore or maybe we would, but we would carry around a separate list of `measurements` and in the end don't return the scenario but return the correct measurements. We could take the return of `Parallel.map` there and simply flatten it and assign the scenario `run_times` to it. Imo that'd make the function "simpler" as it dealt with simpler structures and allow this work.

Happy to give this a shot myself if you're tired of me criticing things :D

Reply:
I'm not totally sure where you're thinking of, since for me the link just shows where we currently have the parallelization. If you want to take a swing at it, go for it, or I can do it if you update the general area you're thinking of moving the parallelization.